src/vstest.console/InProcessVsTestConsoleWrapper.cs

```diff
@@ -88,6 +89,8 @@ public InProcessVsTestConsoleWrapper(ConsoleParameters consoleParameters)
 #endif
         consoleParameters.PortNumber = port;
 
+        UpdateDotnetLookupPath();
```

Comment:
This fixes this particular problem, but makes the behavior different between using in-process console and not. When in-process console is not used, we are given a full set of env variables when wrapper is instantiated, and we use those to start vstest.console. Testhost process (and other processes started by vstest.console) then inherits just those env variables. 

https://github.com/microsoft/vstest/blob/main/src/Microsoft.TestPlatform.VsTestConsole.TranslationLayer/ConsoleParameters.cs#L42-L55

In-process console should probably do the same, and use those env variables to start additional processes. Otherwise the environment will be different for each usecase.

Reply:
Many thanks for the suggestion ! I've actually looked into this first thing when trying to figure out a way to fix the issue and I agree, it would be ideal if we could fix it like you suggested, but unfortunately it doesn't seem possible (at least from my POV).

The reason is that the in-process wrapper, by definition, runs in process, meaning in the `TestWindowStoreHost` process where there's no isolation we can use like AppDomains since environment variables are set at the process level and this seems to be the most granular option we get. We cannot clear the environment variables inside the in-process wrapper and only set the env variables we get via console parameters because that would mess with the TW process. We get the best of both worlds really by keeping the env vars inherited from the TW and (eventually) overriding them because the env var sets are mostly identical, the only thing being left intact is the env vars exclusive to TW.

Propagating the env var set from console params through TP layers down to testhost and other child processes like data collectors is not trivial and the problem we're trying to solve at this point is more theoretical than a real problem. Because of that I'd rather better use this time someplace else. Moreover, we don't really know what a testhost really is, it can be a separate process but it can be loaded as an extension in vstest.console (as with UWP) and then we get the same issue since the testhost will run as part of the vstest.console process, which in fact will be the same process as `TestWindowStoreHost`. We'd have to convince the UWP folks (and other extension writers on the way) to do this propagation when spawning their own processes if they happen to do this.

All in all, I feel confident we can go with this approach at least for now and fix other issues that may appear on the way rather than undertaking the task of propagating the env vars as outlined above, which is not a bulletproof approach either ...

Reply:
There is existing abstraction over environment, I think you could use that. And make sure it is used everywhere you need it.